lib/hsa/kernels/rfi_chime_input_sum.cl

```diff
@@ -55,16 +66,35 @@ rfi_chime_inputsum(
         uint address = gy + gz * gy_size;
         if (n * N == 0){
             output[address] = -1.0;
+            output_var[address] = -1.0;
             output_mask[address] = 1;
         } else {
             float new_sq_power_across_input = sq_power_across_input[0] * cf * cf;
             float SK = ((n + 1) / (n - 1)) * ((new_sq_power_across_input / (n * N)) - 1);
+
+            // Calculate the truncation bias correction for the SK value
+            const float var = var_across_input[0] * one_over_256;
+            const float var2 = var * var;
+            const float var_sqrt = sqrt((double)var);
+            const float sk_correction = bias_coeffs[0] * var2 * var_sqrt + 
+                                        bias_coeffs[1] * var2 +
+                                        bias_coeffs[2] * var * var_sqrt +
+                                        bias_coeffs[3] * var +
+                                        bias_coeffs[4] * var_sqrt +
+                                        bias_coeffs[5]; 
+            const float bias = 1.0 - sk_correction;
+
+            // Correct SK for truncation bias
+            SK += bias * trunc_bias_switch;
```

Comment:
Like this. 

Note you need to invert the sign on the actual SK correction, but that's actually better as a bias would typically be defined such that you *subtract* to correct it.

```suggestion
            const float bias = bias_coeffs[0] * var2 * var_sqrt + 
                               bias_coeffs[1] * var2 +
                               bias_coeffs[2] * var * var_sqrt +
                               bias_coeffs[3] * var +
                               bias_coeffs[4] * var_sqrt +
                               bias_coeffs[5]; // You'll need to change the definition of this coeff above to subtract 1 from it

            // Correct SK for truncation bias
            SK -= bias * trunc_bias_switch;
```